spring-native-configuration/src/main/java/org/springframework/ValidatedComponentProcessor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework;
+
+import org.springframework.nativex.hint.ProxyBits;
+import org.springframework.nativex.type.ComponentProcessor;
+import org.springframework.nativex.type.NativeContext;
+import org.springframework.nativex.type.Type;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Recognize spring.components that need validate proxies and register them.
+ *
+ * @author Petr Hejl
+ */
+public class ValidatedComponentProcessor implements ComponentProcessor {
+
+	@Override
+	public boolean handle(NativeContext imageContext, String componentType, List<String> classifiers) {
+		Type type = imageContext.getTypeSystem().resolveName(componentType);
+		return type.isAtValidated(true);
+	}
+
+	@Override
+	public void process(NativeContext imageContext, String componentType, List<String> classifiers) {
+		Type type = imageContext.getTypeSystem().resolveName(componentType);
+		List<String> transactionalInterfaces = new ArrayList<>();
```

Comment:
the name of this looks like a leftover from the transactional component processor it was a copy of?

Reply:
Yes. Thanks for catching that. Fixed.